jerry-core/ecma/builtin-objects/ecma-builtin-function-prototype.cpp

```diff
@@ -226,7 +226,151 @@ ecma_builtin_function_prototype_object_bind (ecma_value_t this_arg, /**< this ar
                                              const ecma_value_t* arguments_list_p, /**< list of arguments */
                                              ecma_length_t arguments_number) /**< number of arguments */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, arguments_list_p, arguments_number);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 2. */
+  if (!ecma_op_is_callable (this_arg))
+  {
+    ret_value = ecma_make_throw_obj_completion_value (ecma_new_standard_error (ECMA_ERROR_TYPE));
+  }
+  else
+  {
+    /* 4. 11. 18. */
+    ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_FUNCTION_PROTOTYPE);
+    ecma_object_t *function_p = ecma_create_object (prototype_obj_p, true, ECMA_OBJECT_TYPE_BOUND_FUNCTION);
+
+    ecma_deref_object (prototype_obj_p);
+
+    /* 7. */
+    ecma_property_t *target_function_prop_p;
+    target_function_prop_p = ecma_create_internal_property (function_p,
+                                                            ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_TARGET_FUNCTION);
+
+    target_function_prop_p->u.internal_property.value = ecma_copy_value (this_arg, false);
+
+    /* 8. */
+    ecma_property_t *bound_this_prop_p;
+    bound_this_prop_p = ecma_create_internal_property (function_p, ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS);
+
+    size_t arg_list_size;
+
+    if (arguments_number > 0)
+    {
+      bound_this_prop_p->u.internal_property.value = ecma_copy_value (arguments_list_p[0], false);
+
+      arg_list_size = (size_t) (arguments_number) * sizeof (ecma_value_t);
+    }
+    else
+    {
+      ecma_object_t *glob_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_GLOBAL);
+      bound_this_prop_p->u.internal_property.value = ecma_copy_value (ecma_make_object_value (glob_obj_p), false);
+      ecma_deref_object (glob_obj_p);
```

Comment:
In [10.4.3](http://www.ecma-international.org/ecma-262/5.1/#sec-10.4.3) step 2, but we do it in [L663](https://github.com/Samsung/jerryscript/blob/master/jerry-core/ecma/operations/ecma-function-object.cpp#L663).
So I will set here the value of the `bound_this_prop` to `ECMA_SIMPLE_VALUE_UNDEFINED`. It will be correct?